cpp/src/arrow/dataset/partition.cc

```diff
@@ -395,18 +396,18 @@ Result<std::vector<KeyValuePartitioning::Key>> FilenamePartitioning::ParseKeys(
   return ParsePartitionSegments(segments);
```

Comment:
Shouldn't the line above be...

```
std::vector<std::string> segments =
      fs::internal::SplitAbstractPath(StripNonPrefix(fs::internal::GetAbstractPathParent(path).second)path), kFilenamePartitionSep);
```
In other words, if the path is `foo/bar/x=3_y=5_chunk0.parquet`:

```
StripNonPrefix(path) // foo/bar/x=3_y=5
fs::internal::GetAbstractPathParent(path).second // x=3_y=5_chunk0.parquet
StripNonPrefix(fs::internal::GetAbstractPathParent(path).second) // x=3_y=5
```

Reply:
Yes, corrected that, thanks!